### PR TITLE
refactor: Add support for transaction types

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Implements https://app.clickup.com/t/9002019994/PRF-4903
Implements https://app.clickup.com/t/9002019994/PRF-5027

### CODE REVIEW

File: apps/evm_family/evm_user_verification.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+bool evm_verify_eth_transfer(const evm_txn_context_t *txn_context) {
```
Comment (amanCypherock) [inline apps/evm_family/evm_user_verification.c:106]: Consider changing the name

File: apps/evm_family/evm_user_verification.c [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  snprintf(display, sizeof(display), UI_TEXT_SEND_TXN_FEE, value, unit);
+  if (!core_scroll_page(UI_TEXT_TXN_FEE, display, evm_send_error)) {
+    return status;
+  }
+
+  return true;
+}
+
+bool evm_verify_clear_signing(const evm_txn_context_t *txn_context) {
+  bool status = false;
```
Comment (amanCypherock) [inline apps/evm_family/evm_user_verification.c:159]: We don't really need status variable as every return is deterministic here

File: apps/evm_family/evm_txn_helpers.c [hunk 3]
```diff
@@ -173,38 +174,43 @@ static uint64_t get_decode_length(const uint8_t *seq,
...
-                ETH_UTXN_ABI_DECODE_OK)
-                   ? PAYLOAD_WHITELISTED
-                   : PAYLOAD_SIGNATURE_NOT_WHITELISTED;
+    if (!evm_is_token_whitelisted) {
+      result =
+          (ETH_ExtractArguments(utxn_ptr->payload, utxn_ptr->payload_size) ==
+           ETH_UTXN_ABI_DECODE_OK)
+              ? PAYLOAD_WHITELISTED
+              : PAYLOAD_SIGNATURE_NOT_WHITELISTED;
+    }
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.c:213]: Let's add a todo here for detecting clear sign or blind sign transaction using the function tag value along with function signature.

**Review by amanCypherock — CHANGES_REQUESTED** [review]
At file ev_sign_txn.c line 270 either update the todo and consider using the token name from init txn as of now.

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-5027 Port the blind signing with send transaction](https://app.clickup.com/t/9002019994/PRF-5027)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4903 Integrate EVM tokens transfers](https://app.clickup.com/t/9002019994/PRF-4903)

File: apps/evm_family/evm_user_verification.c (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_user_verification.c:106]: renaming to `evm_verify_transfer`. any suggestions?

File: apps/evm_family/evm_user_verification.c (hunk 2 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_user_verification.c]: removed

File: apps/evm_family/evm_txn_helpers.c (hunk 3 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.c:215]: added

**Comment by ujjwal-cyph** [comment]
> At file ev_sign_txn.c line 270 either update the todo and consider using the token name from init txn as of now.

updated the todo

File: apps/evm_family/evm_user_verification.c (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_user_verification.c]: renamed to `evm_verify_transfer`

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**